[WPE] gamepad crashes on raspberry pi 4 (arm64)

ceyusa opened the pull request:
#### a6099cec8ff21c0ea765f38feae3ecb899ba19b4
<pre>
[WPE] gamepad crashes on raspberry pi 4 (arm64)
<a href="https://bugs.webkit.org/show_bug.cgi?id=248210">https://bugs.webkit.org/show_bug.cgi?id=248210</a>

Reviewed by Michael Catanzaro.

This is a breadcrumbs of the initial merge, because in libwpe uintptr_t is used for
gamepad id, while in webkit&apos;s platform code unsigned is used.

* Source/WebCore/platform/gamepad/libwpe/GamepadProviderLibWPE.cpp:
(WebCore::GamepadProviderLibWPE::gamepadConnected):
(WebCore::GamepadProviderLibWPE::gamepadDisconnected):
(WebCore::GamepadProviderLibWPE::removeGamepadForId):
* Source/WebCore/platform/gamepad/libwpe/GamepadProviderLibWPE.h:

Canonical link: <a href="https://commits.webkit.org/256940@main">https://commits.webkit.org/256940@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1235f845838f5ae4a0ddf42937e1226087c17356

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97287 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6551 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30437 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106806 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167072 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101256 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6847 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35286 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89681 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103487 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102950 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5140 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83915 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32126 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86992 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88813 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75052 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/567 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20285 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/551 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21743 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4783 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5352 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44231 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1795 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41072 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1235f845838f5ae4a0ddf42937e1226087c17356. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/6728

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1235f845838f5ae4a0ddf42937e1226087c17356. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/6728

ceyusa committed:
[WPE] gamepad crashes on raspberry pi 4 (arm64)
https://bugs.webkit.org/show_bug.cgi?id=248210

Reviewed by Michael Catanzaro.

This is a breadcrumbs of the initial merge, because in libwpe uintptr_t is used for
gamepad id, while in webkit's platform code unsigned is used.

* Source/WebCore/platform/gamepad/libwpe/GamepadProviderLibWPE.cpp:
(WebCore::GamepadProviderLibWPE::gamepadConnected):
(WebCore::GamepadProviderLibWPE::gamepadDisconnected):
(WebCore::GamepadProviderLibWPE::removeGamepadForId):
* Source/WebCore/platform/gamepad/libwpe/GamepadProviderLibWPE.h:

Canonical link: https://commits.webkit.org/256940@main

webkit-commit-queue commented:
Committed 256940@main (a6099cec8ff2): <https://commits.webkit.org/256940@main>

Reviewed commits have been landed. Closing PR #6728 and removing active labels.
